src/test/java/seedu/guest/model/guest/GuestTest.java

```diff
@@ -70,5 +74,46 @@ public void equals() {
         // different email -> returns false
         editedAlice = new GuestBuilder(ALICE).withEmail(VALID_EMAIL_BOB).build();
         assertFalse(ALICE.equals(editedAlice));
+
+        // different date range -> returns false
+        editedAlice = new GuestBuilder(ALICE).withDateRange(VALID_DATE_RANGE_BOB).build();
+        assertFalse(ALICE.equals(editedAlice));
+
+        // different number of guests -> returns false
+        editedAlice = new GuestBuilder(ALICE).withNumberOfGuests(VALID_NUMBER_OF_GUESTS_BOB).build();
+        assertFalse(ALICE.equals(editedAlice));
+    }
+
+    @Test
+    public void hashcode() {
+        // same values -> returns true
+        Guest aliceCopy = new GuestBuilder(ALICE).build();
+        assertEquals(ALICE.hashCode(), aliceCopy.hashCode());
+
+        // same object -> returns true
+        assertEquals(ALICE.hashCode(), aliceCopy.hashCode());
+
+        // different person -> returns false
+        assertNotEquals(ALICE.hashCode(), BOB.hashCode());
+
+        // different name -> returns false
+        Guest editedAlice = new GuestBuilder(ALICE).withName(VALID_NAME_BOB).build();
+        assertNotEquals(ALICE.hashCode(), editedAlice.hashCode());
+
+        // different phone -> returns false
+        editedAlice = new GuestBuilder(ALICE).withPhone(VALID_PHONE_BOB).build();
+        assertNotEquals(ALICE.hashCode(), editedAlice.hashCode());
+
+        // different email -> returns false
+        editedAlice = new GuestBuilder(ALICE).withEmail(VALID_EMAIL_BOB).build();
+        assertNotEquals(ALICE.hashCode(), editedAlice.hashCode());
+
+        // different date range -> returns false
+        editedAlice = new GuestBuilder(ALICE).withDateRange(VALID_DATE_RANGE_BOB).build();
+        assertNotEquals(ALICE.hashCode(), editedAlice.hashCode());
+
+        // different email -> returns false
```

Comment:
Should this be different number of guests?

Reply:
Yes it should. Changed.